src/go/k8s/apis/redpanda/v1alpha1/redpanda_clusterspec_types.go

```diff
@@ -115,19 +119,38 @@ type UsersItems struct {
 
 // TLS is a top level field of the values file
 type TLS struct {
-	Certs             *Certs `json:"certs,omitempty"`
-	Enabled           bool   `json:"enabled"`
-	RequireClientAuth bool   `json:"requireClientAuth"`
+	Certs   map[string]*Certificate `json:"certs,omitempty"`
+	Enabled *bool                   `json:"enabled,omitempty"`
+}
+
+type Certificate struct {
+	IssuerRef *IssuerRef     `json:"issuerRef,omitempty"`
+	SecretRef *SecretRef     `json:"secretRef,omitempty"`
+	Duration  *time.Duration `json:"duration,omitempty"`
+	CAEnabled bool           `json:"caEnabled"`
 }
 
-// Certs is a top level field of the values file
-type Certs struct{}
+type IssuerRef struct {
+	Name string `json:"name"`
+	Kind string `json:"kind"`
+}
+
+type SecretRef struct {
+	Name string `json:"name"`
+}
+
+// ListenerTLS is a top level field of the values file
+type ListenerTLS struct {
+	Cert              *string `json:"certs,omitempty"`
+	Enabled           *bool   `json:"enabled,omitempty"`
+	RequireClientAuth bool    `json:"requireClientAuth"`
+}
 
 // External is a top level field of the values file
 type External struct {
 	Addresses   json.RawMessage `json:"addresses,omitempty"`
 	Domain      string          `json:"domain,omitempty"`
-	Enabled     bool            `json:"enabled"`
+	Enabled     bool            `json:"enabled,omitempty"`
```

Comment:
@alejandroEsc this still requires a `true` or `false`, resulting in:

```
* spec.clusterSpec.external.enabled: Required value
```

The Helm chart enables `external` by default.

Should we be using a pointer (`*bool`) like some of the other `omitempty` values so that we can use the Helm defaults?

Reply:
understood, will take care of that